src/cryptonote_core/blockchain.cpp

```
@@ -299,110 +299,399 @@ uint64_t Blockchain::get_current_blockchain_height(bool lock) const
+        }
+        catch (const std::exception &e)
+        {
+          MFATAL("Unable to process block " << block_height << " for updating master node list: " << e.what());
           return false;
         }
-        mnl_iteration_duration += clock::now() - mnl_start;
+        mnl_interval_duration += clock::now() - mnl_start;
       }
 
-      if (m_bns_db.db && (block_height >= bns_height))
+      if (m_bns_db.db)
```

Clang (14.0.6)

[note] +3, including nesting penalty of 2, nesting level increased to 3

(clang)